### PR TITLE
staticcheck follow-up to address late feedback

### PR DESCRIPTION
**Pull request by tallclair**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:

Address @BenTheElder open comments from https://github.com/kubernetes/kubernetes/pull/81189

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

/sig contributor-experience

/assign @BenTheElder

### CODE REVIEW

File: hack/verify-staticcheck.sh [hunk 1]
```diff
@@ -94,7 +92,7 @@ while read -r error; do
 done < <(staticcheck -checks "${checks}" "${all_packages[@]}" 2>/dev/null || true)
 
 export IFS=$'\n'
```
Comment (BenTheElder) [inline hack/verify-staticcheck.sh]: IFS is set to '' in `kube::util::read-array`, is this doing anything?
https://github.com/kubernetes/kubernetes/blob/f17b608157ae94369656d8aebeae76bef916012d/hack/lib/util.sh#L742

Comment (tallclair) [reply hack/verify-staticcheck.sh]: This one is required to properly expand the `really_failing` array. I added a comment to that effect.

I also realized that the IFS set higher up in the file was unused, and removed that one.

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve
thanks!

**Pull request merged**